### ISSUE TITLE
[0.9.0 staging-1245] Web elections: sometimes I lose my session when voting

### ISSUE THREAD

**Issue by beastyn** [StrangeLoopGames/EcoIssues#13661]
Im on  my web server page. I can browse through pages, read info. But when Im trying to submit a vote fo the law, I`ve got the error message "Error: You are not logged in"
![bandicam 2019-11-12 15-55-24-188](https://user-images.githubusercontent.com/27898520/68666011-e9774e00-0564-11ea-94f5-0f1fc7a6a45b.jpg)

Page updating won't help, but if Im clicking on the Vote in-game, everything will be fine on the new opened page.

**Comment by beastyn** [comment]
It appears if cleant was rebooted, but page is from the previous session.

**Comment by j3fffff** [comment]
@denysaw could we store unsubmitted user input info in a session variable or something?

**Comment by denysaw** [comment]
@j3fffff We could store it in LocalStorage, sure. Just don't get how it's connected to the task :) Task is about changing `authtoken` probably by logging in-game in parallel and then getting back to voting. In such or similar case, surely World on any API call will return `You are not logged in`.

**Comment by j3fffff** [comment]
I was thinking as a solution, if we save their input, then they can relogin and submit their law or whatever.

**Comment by denysaw** [comment]
For comment we can do that. For vote it's usually Yes/No and better to not store Candidates position as user would think he already voted.

**Comment by denysaw** [comment]
Yeah, for laws, comments etc we could store everything in localStorage, yep

**Comment by j3fffff** [comment]
updated my response ^
agreed on candidates

**Comment by denysaw** [comment]
Yep, in separate issue then
